Modify mantid FilterEvents algorithm

jmborr opened the pull request:
**References**  
[187: [Enabler] Modify mantid FilterEvents algorithm #968](https://ornlrse.clm.ibmcloud.com/ccm/resource/itemOid/com.ibm.team.workitem.WorkItem/_TyAQII1DEe2anfFdXJwBcA)

**Description of work.**  
- [x] [723: Migrate to splitting events with a TimeSplitter object](https://ornlrse.clm.ibmcloud.com/ccm/resource/itemOid/com.ibm.team.workitem.WorkItem/_FxGXQL3BEe2J9vc1b21wEQ)
- [x] [737: Convert table workspace input to TimeSplitter](https://ornlrse.clm.ibmcloud.com/ccm/resource/itemOid/com.ibm.team.workitem.WorkItem/_Pw9GwMGfEe2J9vc1b21wEQ)
- [x] [738: Convert SplittersWorkspace into TimeSplitter](https://ornlrse.clm.ibmcloud.com/ccm/resource/itemOid/com.ibm.team.workitem.WorkItem/_SxdqYMGfEe2J9vc1b21wEQ)
- [x] [739: Convert MatrixWorkspace into TimeSplitter](https://ornlrse.clm.ibmcloud.com/ccm/resource/itemOid/com.ibm.team.workitem.WorkItem/_U-wBUMGfEe2J9vc1b21wEQ)
- [x] [1014: Implement TimeSplitter unit tests for non-numerical and shifted targets](https://ornlrse.clm.ibmcloud.com/ccm/web/projects/Neutron%20Data%20Project%20%28Change%20Management%29#action=com.ibm.team.workitem.viewWorkItem&id=1014)
- [x] [1094: Update Algorithm documentation](https://ornlrse.clm.ibmcloud.com/ccm/web/projects/Neutron%20Data%20Project%20(Change%20Management)#action=com.ibm.team.workitem.viewWorkItem&id=1094)
- [x]  [1185: Add FilterEvents tests for FilterStartTime](https://ornlrse.clm.ibmcloud.com/ccm/web/projects/Neutron%20Data%20Project%20%28Change%20Management%29#action=com.ibm.team.workitem.viewWorkItem&id=1185)

**TODO's to resolve _before merging the PR_**
- [x]  `FilterEvents::examineAndSortEventWS()` [sorts all `EventList`'s in the input workspace](https://github.com/mantidproject/mantid/blob/main/Framework/Algorithms/src/FilterEvents.cpp#L359) but this seems unnecessary because `TimeSplitter.splitEventList() sorts`  each `EventList`
- [x] Clone the input log properties into the output workspace. The prototype workspace `prototype_ws` should contain the logs of the input workspace, minus the `ExcludeSpecifiedLogs`. Then the cloning step `optws = prototype_ws->clone()` should also clone the logs seamlessly.

**To test:**

<!-- Instructions for testing. -->

Refs #34794.

<!-- delete this if you added release notes
*This does not require release notes* because **fill in an explanation of why**
If you add release notes please save them as a separate file using the Issue or PR number as the file name. Check the file is located in the correct directory for your note(s).
-->

<!-- Ensure the base of this PR is correct (e.g. release-next or main)
Finally, don't forget to add the appropriate labels, milestones, etc.!  -->

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

bolotovskyr committed:
resolve conflicts with branch main

bolotovskyr committed:
Resolve conflicts with branch main

jmborr committed:
reorganize input properties

Signed-off-by: Jose Borreguero <borreguero@gmail.com>

jmborr committed:
sanitize FilterEventsTest::createEventWorkspace

Signed-off-by: Jose Borreguero <borreguero@gmail.com>

bolotovskyr committed:
Handle cases with TimeROI returning multiple SplittingInterval-s

jmborr committed:
add comments to createSplittersWorkspace

Signed-off-by: Jose Borreguero <borreguero@gmail.com>

jmborr committed:
test for addjacent boundaries

Signed-off-by: Jose Borreguero <borreguero@gmail.com>

jmborr committed:
description of the splitted workspace

Signed-off-by: Jose Borreguero <borreguero@gmail.com>

jmborr committed:
add TimeSplitters::toSplitters() method

Signed-off-by: Jose Borreguero <borreguero@gmail.com>

bolotovskyr committed:
Use TimeSplitter in FilterEvents::convertSplittersWorkspaceToVectors.

jmborr committed:
add test case in test_toSplitters

Signed-off-by: Jose Borreguero <borreguero@gmail.com>

jmborr committed:
add filterEventsBySplittersSerial( )

Signed-off-by: Jose Borreguero <borreguero@gmail.com>

jmborr committed:
fix bug in TimeSplitter::SplitEventVec

Signed-off-by: Jose Borreguero <borreguero@gmail.com>

bolotovskyr committed:
Improve handling of descriptiveNames case. Better comments.

bolotovskyr committed:
More clean-up.

jmborr committed:
test for leaping events over splitters

Signed-off-by: Jose Borreguero <borreguero@gmail.com>

bolotovskyr committed:
Address cpp check and fix EventList::splitByPulseTime.

bolotovskyr reviewed (approved):
Looks straightforward.

jmborr committed:
remove redundant sorting of events

Events are sorted prior to splitting them into the destinations workspaces. Thus, this is unnecessary

Signed-off-by: Jose Borreguero <borreguero@gmail.com>

jmborr committed:
Parallelize filterEventsBySplitters

Parallelize filterEventsBySplittersSerial and delete old filterEventsBySplitters

Signed-off-by: Jose Borreguero <borreguero@gmail.com>

jmborr committed:
Rename examineAndSortEventWS

The member function is not doing any sorting of events

Signed-off-by: Jose Borreguero <borreguero@gmail.com>

jmborr committed:
output names of tests for easier inspection

Signed-off-by: Jose Borreguero <borreguero@gmail.com>

bolotovskyr committed:
Use TimeSplitter for TableWorkspace input in FilterEvents

jmborr committed:
delete (most) of the functionality splitting the logs

Signed-off-by: Jose Borreguero <borreguero@gmail.com>

jmborr committed:
comment out failing log assertions

Signed-off-by: Jose Borreguero <borreguero@gmail.com>

jmborr committed:
address cppcheck complaints

Signed-off-by: Jose Borreguero <borreguero@gmail.com>

jmborr committed:
fix test test_FilterNoCorrection

Signed-off-by: Jose Borreguero <borreguero@gmail.com>

jmborr committed:
FilterEventsTest::test_FilterWOCorrection2 bugfix

Signed-off-by: Jose Borreguero <borreguero@gmail.com>

jmborr committed:
bugfix FilterEventsTest::test_FilterWithCustumizedCorrection

Signed-off-by: Jose Borreguero <borreguero@gmail.com>

bolotovskyr committed:
TimeSplitter for MatrixWorkspace input, OutputUnfilteredEvents.

jmborr committed:
output ROI as intersection of original and splitted

Signed-off-by: Jose Borreguero <borreguero@gmail.com>

bolotovskyr committed:
Consolidate input splitter parsers. Update properties help.

bolotovskyr committed:
Add TimeSplitter test for non-numeric targets.

bolotovskyr committed:
Address doxygen complaints.

bolotovskyr committed:
Address doxygen complaints - contd

bolotovskyr committed:
Add TimeSplitter test for shifting targets. More doxygen temp fixes.

jmborr committed:
make invisible property SplitSampleLogs, and don't use it

Signed-off-by: Jose Borreguero <borreguero@gmail.com>

bolotovskyr committed:
Implement TimeSplitter::addLastTime. Clean-up.

jmborr committed:
turn the leanding and trailing masks into ROI for unfiltered workspace

Signed-off-by: Jose Borreguero <borreguero@gmail.com>

jmborr committed:
bugfix tests checkin for the unfiltered workspace

Signed-off-by: Jose Borreguero <borreguero@gmail.com>

jmborr committed:
bugfix createEventWorkspaceElastic and createMatrixSplittersElastic

Signed-off-by: Jose Borreguero <borreguero@gmail.com>

jmborr committed:
bugfix test_FilterDGCorrection

Signed-off-by: Jose Borreguero <borreguero@gmail.com>

jmborr committed:
bugfix test_FilterIndirectGeometryCorrection

Signed-off-by: Jose Borreguero <borreguero@gmail.com>

jmborr committed:
bugfix test_FilterRelativeTime

Signed-off-by: Jose Borreguero <borreguero@gmail.com>

jmborr committed:
Remove uses of TimeSeriesPropertyLogs and ExcludeSpecifiedLogs

Signed-off-by: Jose Borreguero <borreguero@gmail.com>

bolotovskyr committed:
Fix doc tests missing unfiltered wkspace. Clean-up.

jmborr committed:
Make TimeSeriesPropertyLogs and ExcludeSpecifiedLogs invisible

Signed-off-by: Jose Borreguero <borreguero@gmail.com>

jmborr committed:
warn of deprecated properties

Signed-off-by: Jose Borreguero <borreguero@gmail.com>

jmborr committed:
clean up DEBUG: marks

Signed-off-by: Jose Borreguero <borreguero@gmail.com>

jmborr committed:
add proton charge test

Signed-off-by: Jose Borreguero <borreguero@gmail.com>

jmborr committed:
enforce output of unfiltered workspace

Signed-off-by: Jose Borreguero <borreguero@gmail.com>

jmborr committed:
remove variable assignment

Signed-off-by: Jose Borreguero <borreguero@gmail.com>

peterfpeterson commented on Framework/Algorithms/src/FilterEvents.cpp:
```diff
@@ -128,69 +175,54 @@ void FilterEvents::init() {
-                  "If selected, the names of the output workspaces will "
-                  "include information about each slice.");
+  declareProperty("OutputUnfilteredEvents", false, "If selected, unfiltered events will be output.");
+  setPropertyGroup("OutputUnfilteredEvents", titleOutputWksp);
```
These strings being copied everywhere reminds me of our discussion of a local `PropertyNames` namespace from the other day. Good times!

jmborr replied:
Oh no!, not another Holy war :anguished: :exclamation:

peterfpeterson commented on Framework/Algorithms/src/FilterEvents.cpp:
```diff
@@ -128,69 +175,54 @@ void FilterEvents::init() {
 
 std::map<std::string, std::string> FilterEvents::validateInputs() {
-  const std::string SPLITER_PROP_NAME = "SplitterWorkspace";
+  const std::string SPLITTER_PROP_NAME = "SplitterWorkspace";
```
Here it is again!

peterfpeterson commented on Framework/Algorithms/src/FilterEvents.cpp:
```diff
@@ -128,69 +175,54 @@ void FilterEvents::init() {
+  setPropertyGroup("OutputUnfilteredEvents", titleOutputWksp);
 }
 
 std::map<std::string, std::string> FilterEvents::validateInputs() {
```
Should this method verify that the user only specified the filtering in a single way? It is more of a "nice to have" than a "must be done."

bolotovskyr replied:
Could you please clarify what you mean by a "single way"? Are you referring to the 3 supported types of splitter workspaces - Splitters, Table, Matrix?  In that case the answer would be: `OutputUnfilteredEvents` should work the same way regardless of how the splitter was specified.

bolotovskyr replied:
Sorry, I misunderstood your comment.  I wasn't looking at the last line of the code snippet!  Yes, that would be a nice check to have.  I also have the following question.  Currently `FilterEvents::validateInputs` checks things like: a table workspace must have 3 columns, a matrix workspace must be of histogram type, etc.  But now we have the same kind of checks in the corresponding `TimeSplitter` constructors.  Should we then delete these checks from `FilterEvents::validateInputs`?

peterfpeterson replied:
That's a bit more subtle. `validateInputs()` will put error messages in the gui before the algorithm starts doing its thing, but the constructors should definitely keep there checking. One way to leverage the constuctors is to use them in `validateInputs()` to set member variables that are used in the execute. I both like and dislike that method, but it does reduce duplicate code.

bolotovskyr replied:
@peterfpeterson, I am back wondering what a "single way" means.  The user always provides just a single workspace describing splitters.  We then cast it to 3 different types of  workspace, and whichever pointer is valid - we then use it.

bolotovskyr replied:
> One way to leverage the constuctors is to use them in validateInputs() to set member variables that are used in the execute. I both like and dislike that method, but it does reduce duplicate code.

bolotovskyr replied:
One way to leverage the constuctors is to use them in validateInputs() to set member variables that are used in the execute. I both like and dislike that method, but it does reduce duplicate code.

peterfpeterson replied:
I think it is actually covered. It was a comment I put in before I had read through more of the code.

bolotovskyr replied:
One way to leverage the constuctors is to use them in validateInputs() to set member variables that are used in the execute. I both like and dislike that method, but it does reduce duplicate code.

In this case I would rather not mix input validation with object initialization. I suggest to keep this "code duplication" as is.

jmborr committed:
Resolve conflicts with branch main

The code doesn't use the destination-workspace index of the SplittingInterval, thus it doesn't make sense using this type

Signed-off-by: Jose Borreguero <borreguero@gmail.com>

jmborr committed:
remove unsynced docstring

Signed-off-by: Jose Borreguero <borreguero@gmail.com>

jmborr committed:
remove orphan docstring

Signed-off-by: Jose Borreguero <borreguero@gmail.com>

bolotovskyr committed:
MatrixWorkspace doctest. Support TableWorkspace float column type.

bolotovskyr committed:
Correct indent size.

peterfpeterson commented on Framework/Algorithms/src/FilterEvents.cpp:
```diff
@@ -477,17 +484,27 @@ void FilterEvents::processAlgorithmProperties() {
+  //-------------------------------------------------------------------------
+  bool splitSampleLogs = getProperty("SplitSampleLogs");
+  if (splitSampleLogs == false)
+    g_log.warning() << "Option SplitSampleLogs is deprecated. Logs will be split\n";
```
Can we add a comment/word about this property being ignored?

jmborr replied:
What is there missing that the log warning "explanation" is not addressing?

peterfpeterson replied:
I think `"Option SplitSampleLogs is deprecated and ignored. Logs will be split\n"` is that little bit more explicit.

peterfpeterson commented on Framework/Algorithms/src/FilterEvents.cpp:
```diff
@@ -477,17 +484,27 @@ void FilterEvents::processAlgorithmProperties() {
+
+  std::vector<std::string> timeSeriesPropertyLogs = getProperty("TimeSeriesPropertyLogs");
+  if (!timeSeriesPropertyLogs.empty())
+    g_log.warning() << "Options TimeSeriesPropertyLogs and ExcludeSpecifiedLogs are deprecated. "
```
This one too

jmborr replied:
same question as before :yum:

peterfpeterson commented on Framework/Algorithms/src/FilterEvents.cpp:
```diff
@@ -521,620 +538,183 @@ void FilterEvents::groupOutputWorkspace() {
-    new_property->setUnits(tsp->units());
-    output_vector.emplace_back(std::move(new_property));
-  }
+void FilterEvents::parseInputSplitters() {
```
This is so much clearer

peterfpeterson commented on Framework/Algorithms/src/FilterEvents.cpp:
```diff
@@ -521,620 +538,183 @@ void FilterEvents::groupOutputWorkspace() {
-  // 3. Order if not ordered and add workspace for events excluded
-  if (!inorder) {
-    std::sort(m_splitters.begin(), m_splitters.end());
+  size_t min_expected_number_of_indexes =
```
This appears to be constant. If that is correct, mark it `const` to make it clearer. Otherwise, the comment should be modified to explain how/when it is changed.

peterfpeterson commented on Framework/Algorithms/src/FilterEvents.cpp:
```diff
@@ -1143,17 +723,18 @@ void FilterEvents::createOutputWorkspacesSplitters() {
-      std::stringstream propertynamess;
-      if (wsgroup == -1) {
-        propertynamess << "OutputWorkspace_unfiltered";
+      // Generate the name of the output property
```
```suggestion
      // Generate the name of the output property. Only workspaces that are set as output properties get history added to them
```

jmborr replied:
What do you mean with "history". Is this in the picture?
![Capture](https://user-images.githubusercontent.com/1136006/230919139-0434c6d4-803f-4e51-a121-831ff40c8e24.PNG)

peterfpeterson replied:
Correct. Workspaces that aren't labeled through output parameters in an algorithm, don't get workspace history set. This is a clarification on the comment so developers understand why it is being done

peterfpeterson commented on Framework/Algorithms/test/FilterEventsTest.h:
```diff
@@ -72,6 +73,7 @@ class FilterEventsTest : public CxxTest::TestSuite {
    * (5) "Experiment": 5 pulse times.  10 events in each pulse
    */
   void test_CreatedEventWorskpaceAndSplitter() {
+    std::cout << std::endl << "test_CreatedEventWorskpaceAndSplitter..." << std::endl;
```
remove or convert to mantid logging

peterfpeterson commented on Framework/Algorithms/test/FilterEventsTest.h:
```diff
@@ -91,14 +93,79 @@ class FilterEventsTest : public CxxTest::TestSuite {
 
-  //----------------------------------------------------------------------------------------------
+  void test_protonCharge() {
+    std::cout << std::endl << "test_protonCharge..." << std::endl;
```
remove or convert to mantid logging

peterfpeterson commented on Framework/Algorithms/test/FilterEventsTest.h:
```diff
@@ -117,15 +184,27 @@ class FilterEventsTest : public CxxTest::TestSuite {
-    int64_t runstart_i64 = 20000000000;
-    int64_t pulsedt = 100 * 1000 * 1000;
-    int64_t tofdt = 10 * 1000 * 1000;
+    std::cout << std::endl << "test_FilterNoCorrection..." << std::endl;
```
remove or convert to mantid logging

peterfpeterson commented on Framework/Algorithms/test/FilterEventsTest.h:
```diff
@@ -419,11 +521,28 @@ class FilterEventsTest : public CxxTest::TestSuite {
    */
   void test_FilterElasticCorrection() {
-    EventWorkspace_sptr ws = createEventWorkspaceElastic(0, 1000000);
+    std::cout << std::endl << "test_FilterElasticCorrection..." << std::endl;
```
remove or convert to mantid logging

peterfpeterson commented on Framework/Algorithms/test/FilterEventsTest.h:
```diff
@@ -476,10 +596,14 @@ class FilterEventsTest : public CxxTest::TestSuite {
    */
   void test_FilterDGCorrection() {
-    EventWorkspace_sptr ws = createEventWorkspaceDirect(0, 1000000);
+    std::cout << std::endl << "test_FilterDGCorrection..." << std::endl;
```
remove or convert to mantid logging

peterfpeterson commented on Framework/Algorithms/test/FilterEventsTest.h:
```diff
@@ -526,11 +650,14 @@ class FilterEventsTest : public CxxTest::TestSuite {
   void test_FilterIndirectGeometryCorrection() {
-    // Create workspaces for filtering
-    EventWorkspace_sptr ws = createEventWorkspaceInDirect(0, 1000000);
+    std::cout << std::endl << "test_FilterIndirectGeometryCorrection..." << std::endl;
```
remove or convert to mantid logging

peterfpeterson commented on Framework/Algorithms/test/FilterEventsTest.h:
```diff
@@ -607,6 +734,7 @@ class FilterEventsTest : public CxxTest::TestSuite {
         979: 4: 3.65e+08  -  4.65e+08: 2
    */
   void test_FilterRelativeTime() {
+    std::cout << std::endl << "test_FilterRelativeTime..." << std::endl;
```
remove or convert to mantid logging

peterfpeterson commented on Framework/Algorithms/test/FilterEventsTest.h:
```diff
@@ -895,80 +984,12 @@ class FilterEventsTest : public CxxTest::TestSuite {
    * @brief test_ThrowSameName
    */
   void test_ThrowSameName() {
+    std::cout << std::endl << "test_ThrowSameName..." << std::endl;
```
remove or convert to mantid logging

peterfpeterson commented on Framework/Algorithms/test/FilterEventsTest.h:
```diff
@@ -1007,6 +1027,7 @@ class FilterEventsTest : public CxxTest::TestSuite {
    * @brief test_ThrowSameName
    */
   void test_groupWorkspaces() {
+    std::cout << std::endl << "test_groupWorkspaces..." << std::endl;
```
remove or convert to mantid logging

peterfpeterson commented on Framework/Algorithms/test/FilterEventsTest.h:
```diff
@@ -1044,6 +1064,7 @@ class FilterEventsTest : public CxxTest::TestSuite {
    * basename_startTime_stopTime
    */
   void test_descriptiveWorkspaceNamesTime() {
+    std::cout << std::endl << "test_descriptiveWorkspaceNamesTime..." << std::endl;
```
remove or convert to mantid logging

peterfpeterson commented on Framework/Algorithms/test/FilterEventsTest.h:
```diff
@@ -1105,6 +1127,7 @@ class FilterEventsTest : public CxxTest::TestSuite {
    * basename_startTime_stopTime
    */
   void test_descriptiveWorkspaceNamesLog() {
+    std::cout << std::endl << "test_descriptiveWorkspaceNamesLog..." << std::endl;
```
remove or convert to mantid logging

peterfpeterson commented on Framework/Kernel/src/TimeSeriesProperty.cpp:
```diff
@@ -509,8 +509,11 @@ void TimeSeriesProperty<TYPE>::splitByTimeVector(const std::vector<DateAndTime>
   if (timeToFilterTo.size() != inputWorkspaceIndicies.size() + 1) {
-    throw std::runtime_error("Input time vector's size does not match(one more larger than) target "
-                             "workspace index vector's size inputWorkspaceIndicies.size() \n");
+    // throw std::runtime_error("Input time vector's size does not match(one more larger than) target "
```
Remove commented out code

jmborr replied:
@bolotovskyr Git blame shows it's you who commented this code. Can you say why? I couldn't find any use of this method anywhere except the unit tests so I have no clue how and when it should be used. Lets me to wonder if we should remove this method!

bolotovskyr replied:
@jmborr That was an oversight.  I commented it out while debugging a unit test and forgot to uncomment back it after I fixed the test.

bolotovskyr replied:
What about `TimeSeriesPropertyTest.h` where `splitByTimeVector()` is used? Are we removing it?

jmborr replied:
we have to check this a bit more. I don't understand in which context we're actually splitting a `TimeSeriesProperty `because the whole point of the `TimROI `is to stop splitting the `TimeSeriesProperty  `objects 🤔

peterfpeterson replied:
Please add comments about that to [EWM742](https://ornlrse.clm.ibmcloud.com/ccm/web/projects/Neutron%20Data%20Project%20(Change%20Management)#action=com.ibm.team.workitem.viewWorkItem&id=742). It should be a separate PR from this

peterfpeterson reviewed (changes requested):
Overall, this is a very impressive body of work. Lots of minor/cosmetic requests.

jmborr committed:
make min_expected_number_of_indexes a constexpr

Signed-off-by: Jose Borreguero <borreguero@gmail.com>

bolotovskyr committed:
Use TimeROI::toTimeIntervals() instead of TimeROI::toSplitters().

bolotovskyr committed:
Clarify comment and error message.

peterfpeterson commented on Framework/Algorithms/test/FilterEventsTest.h:
```diff
@@ -91,14 +93,79 @@ class FilterEventsTest : public CxxTest::TestSuite {
+
+    std::vector<std::string> suffixes{"0", "1", "2", "unfiltered"};
+    std::vector<double> protonCharges;
+    for (const std::string suffix : suffixes) {
```
The osx warning created can be solved by making this
```suggestion
    for (const std::string &suffix : suffixes) {
```

bolotovskyr committed:
Simplify splitter validation. No from1 for non-numeric targets.

jmborr committed:
switch from standard output to logger

Signed-off-by: Jose Borreguero <borreguero@gmail.com>

bolotovskyr commented:
The `std::cout` statements in `FilterEventsTest.h` seem to be useful, especially for debugging, as they mark the beginning of each test's output.  When we switch from `std::cout` to` g_log.notice()`, there is a conflict between what the test's base class does and what our logger does.  So a clean looking output like this:
```
Running 15 tests
test_Initialization...
.
test_CreatedEventWorskpaceAndSplitter...
.
test_protonCharge...  
```
suddenly looks like this:
```
Running 15 testsFilterEventsTest-[Notice] 
FilterEventsTest-[Notice] test_Initialization...
.FilterEventsTest-[Notice] 
FilterEventsTest-[Notice] test_CreatedEventWorskpaceAndSplitter...
.FilterEventsTest-[Notice]
FilterEventsTest-[Notice] test_protonCharge...
```
where the first line looks especially unpleasant.  So I suggest we either keep `std::cout` statements as is or comment them out. Also note that if we search `Mantid` for `std::cout` in `*Test.h` files, we get lots of hits.

jmborr commented:
> I suggest we either keep `std::cout` statements or comment them out. Also note that if we search `Mantid` for `std::cout` in all `*Test.h` files, we get lots of hits.
I don't have any preference between `cout` and `g_log` but I do want these lines output so that I can know when each test begins.

bolotovskyr committed:
Further clean-up. New TimeSplitter constructor.

jmborr committed:
unitialize temp variable

Signed-off-by: Jose Borreguero <borreguero@gmail.com>

jmborr committed:
update golden data for MDNormHYSPECTest.py

Signed-off-by: Jose Borreguero <borreguero@gmail.com>

jmborr committed:
update golden data for HYSPECReductionTest.py

Signed-off-by: Jose Borreguero <borreguero@gmail.com>

jmborr committed:
make temp local to the try/catch block

Signed-off-by: Jose Borreguero <borreguero@gmail.com>

bolotovskyr committed:
Update FilterEvent docs. Improve support for FilterStartTime.

bolotovskyr committed:
Add FilterStartTime tests. Use GPS_EPOCH.

bolotovskyr committed:
Optimize last change to FilterEvents::partialROI

jmborr committed:
simplify test and new golden data

Signed-off-by: Jose Borreguero <borreguero@gmail.com>

jmborr committed:
m_filterStartTime is always initialized

Signed-off-by: Jose Borreguero <borreguero@gmail.com>

jmborr committed:
delete splitByPulseTimeWithMatrix and splitByPulseTimeWithMatrixHelper

Signed-off-by: Jose Borreguero <borreguero@gmail.com>

bolotovskyr committed:
Address coverity warnings.

jmborr committed:
Restore initialization of m_isSplittersRelativeTime

Signed-off-by: Jose Borreguero <borreguero@gmail.com>

jmborr committed:
Save new golden data

Signed-off-by: Jose Borreguero <borreguero@gmail.com>

jmborr committed:
expunge splitByFullTime(TimeSplitter, map<int, EventList *>, bool ,double, double)

Signed-off-by: Jose Borreguero <borreguero@gmail.com>

jmborr committed:
Expunge void splitByPulseTime(Kernel::SplittingIntervalVec &splitter, std::map<int, EventList *> outputs)

Signed-off-by: Jose Borreguero <borreguero@gmail.com>

jmborr committed:
Expunge void splitByPulseTime(const TimeSplitter &splitter, std::map<int, EventList *> outputs) const;

Signed-off-by: Jose Borreguero <borreguero@gmail.com>

jmborr committed:
Expunge splitByPulseTimeHelper()

Signed-off-by: Jose Borreguero <borreguero@gmail.com>

jmborr committed:
Expunge splitByPulseTimeHelper()

Signed-off-by: Jose Borreguero <borreguero@gmail.com>

jmborr committed:
Expunge integrateHelper()

Signed-off-by: Jose Borreguero <borreguero@gmail.com>

jmborr committed:
Expunge EventList::splitByFullTime()

Signed-off-by: Jose Borreguero <borreguero@gmail.com>

jmborr committed:
Expunge TimeSeriesProperty::splitByTimeVector

Signed-off-by: Jose Borreguero <borreguero@gmail.com>

jmborr committed:
Expunge EventList::splitByFullTimeMatrixSplitter

Signed-off-by: Jose Borreguero <borreguero@gmail.com>

jmborr committed:
Expunge EventList::splitByFullTimeHelper

Signed-off-by: Jose Borreguero <borreguero@gmail.com>

jmborr committed:
Expunge EventList::splitByFullTimeVectorSplitterHelper

Signed-off-by: Jose Borreguero <borreguero@gmail.com>

jmborr committed:
Expunge EventList::EventList::splitByFullTimeSparseVectorSplitterHelper

Signed-off-by: Jose Borreguero <borreguero@gmail.com>

jmborr committed:
remove unused forward declarations and headers

Signed-off-by: Jose Borreguero <borreguero@gmail.com>

jmborr committed:
remove unused header

Signed-off-by: Jose Borreguero <borreguero@gmail.com>

jmborr committed:
Expunge EventList::filterByTimeAtSample and EventList::filterByTimeAtSampleHelper

Signed-off-by: Jose Borreguero <borreguero@gmail.com>

jmborr committed:
rearrange methods for easy jump between the methods and their helper methods

Signed-off-by: Jose Borreguero <borreguero@gmail.com>

bolotovskyr committed:
Fix incorrect comment and exception text.

jmborr committed:
copy the properties but filter them according to a TimeROI

Signed-off-by: Jose Borreguero <borreguero@gmail.com>

jmborr committed:
include cppcheck supressions

Signed-off-by: Jose Borreguero <borreguero@gmail.com>

jmborr committed:
inserting replies to reviewer's comments

Signed-off-by: Jose Borreguero <borreguero@gmail.com>

bolotovskyr committed:
Fix handling of single-value cache in ROI-based filtering methods.

bolotovskyr committed:
Fix unmatched cppcheck-suppress issues.

bolotovskyr committed:
Implement LogManager::copyProperties method.

bolotovskyr committed:
Update LogManager::setTimeROI

peterfpeterson commented on Framework/Algorithms/src/FilterEvents.cpp:
```diff
@@ -521,620 +519,181 @@ void FilterEvents::groupOutputWorkspace() {
+
+    // discard log entries outside the ROI
+    optws->mutableRun().copyAndFilterProperties(m_eventWS->run(), roi);
+    optws->mutableRun().setTimeROI(roi);
```
The `copyAndFilterProperties()` method already sets the TimeROI. This line is not needed.

bolotovskyr replied:
The problem is that `copyAndFilterProperties` is implemented in the `PropertyManager `class, while `setTimeROI `is implemented in both `PropertyManager `and `Run `classes.  So in this case:

`optws->mutableRun().setTimeROI(roi)` 

calls the `Run `version, which not only sets the `ROI`, but also calls `integrateProtonCharge `and `setDuration`.  Perhaps we need to move the `copyAndFilterProperties` method to the `Run` class?  Then, indeed, we wouldn't have to call `SetTimeROI` after calling `copyAndFilterProperties`.

bolotovskyr replied:
Update: the problem is solved by adding an optional boolean:

`virtual void setTimeROI(const Kernel::TimeROI &timeroi, const bool applyFilter = true);`

and calling `setTimeROI` from `PropertyManager::copyAndFilterProperties`

`this->setTimeROI(timeROI, false);`

peterfpeterson reviewed (changes requested):
`TimeSplitter::splitEventList` and `TimeSplitter::splitEventVec` should pass the partials by reference to save a copy. This does save a bit of tie b/c you're not creating a copy of the map of pointers

bolotovskyr committed:
Updates to TimeROI-based filtering methods.

bolotovskyr committed:
Empty-Commit

bolotovskyr committed:
Empty-Commit

bolotovskyr committed:
New TimeSeriesProperty constructor. Use TimeROI intersection.

peterfpeterson committed:
Add test to show bug in calculating mathematical mean with TimeROI

Also uncover bug in lastValue(TimeROI) and fix it

peterfpeterson committed:
Fix filteredValuesAsVector

peterfpeterson committed:
Fixup filteredTimesAsVector

peterfpeterson committed:
Update correct values for tests

peterfpeterson committed:
Quiet compiler warning

peterfpeterson committed:
Get rid of infinite call stack when called from FilteredTSP

peterfpeterson committed:
Bring back old behavior

peterfpeterson committed:
Make sure TimeROI does something to use it

peterfpeterson committed:
Update correct value

peterfpeterson committed:
Merge remote-tracking branch 'upstream/ewm1355_mean_math' into ewm_737_738_739

peterfpeterson committed:
Revert to main's copy of the reference file

bolotovskyr committed:
Decouple setTimeROI and removeDataOutsideTimeROI.

peterfpeterson committed:
Make clients explicitly remove values outside of TimeROI

bolotovskyr committed:
Fix doxygen warning.

bolotovskyr committed:
Fix LogManager test.

peterfpeterson committed:
Pass by reference

jmborr committed:
update golden data

Signed-off-by: Jose Borreguero <borreguero@gmail.com>

peterfpeterson committed:
Change names of timers
